enhancements/cluster-api/installing-openshift-natively-via-cluster-api.md

```diff
@@ -0,0 +1,598 @@
+They will be expected to create the same resources as the installer would have created, but with their own configuration.
+
+There are 2 ways to achieve this:
+* By using the installer and customising the Cluster API resources generated by the installer.
+* By manually crafting the infrastructure resources and applying them to the Cluster API control plane, for example, for externally supported platforms.
+
+In the second case, where the installer does not have an opinionated set of Cluster API resources to generate, the installer will generate all resources required except for the `InfrastructureCluster` and `InfrastructureMachineTemplate`.
+It is expected that the user will provide these two resources and update the references in the `Cluster` and `OpenShiftControlPlane` resources to match.
+
+##### Bring your own infrastructure
+
+Cluster API allows users to bring their own infrastructure.
+The various providers allow configuration of existing infrastructure, such as VPCs, subnets etc, and will supplement any missing infrastructure with new infrastructure.
+
+If users wish to bring their own infrastructure, they will be expected to create the required infrastructure on the cloud provider, to their needs, and update the `InfrastructureCluster` and `InfrastructureMachineTemplate` resources to match.
+When configured with bring your own infrastructure, the InfrastructureProvider will not modify the existing infrastructure, but will populate the `InfrastructureCluster` with details of the infrastructure that already exists, allowing operation of the rest of the system.
+
+#### Hive integration
+
+TODO
+
+#### Variation and form factor considerations 
+
+##### Interaction with Machine API
+
+Until the Machine API supports migration to Cluster API, the installer will continue to provision Machine API resources into the guest clusters created by Cluster API.
+
+For temporary Cluster API control planes, the Cluster API resources will be discarded once the cluster is ready and the resulting cluster will look exactly as it does today.
+When the Machine API to Cluster API migration is supported, the installer will be updated to include an option to move the Cluster API resources into the guest cluster creating a joined or self managed cluster.
+
+For permanent Cluster API control planes (e.g. Hive or mulit-cluster use cases), Machine API resources should be dropped and Cluster API resources should be used in the long term to manage the cluster.
+This will mean Hive creating Cluster API MachineDeployments instead of MachineAPI MachineSets.
+
+### API Extensions
+
+#### OpenShiftControlPlane
+
+The `OpenShiftControlPlane` resource will be the configuration for the control plane provider implementation for Cluster API.
+It must adhere to the upstream Cluster API [control plane provider API contract][control-plane-api-contract].
```

Comment:
How much of what's described below is part of that contract versus unique to OpenShift?

Reply:
The entirety of `machineTemplate` is part of the contract, I will try to make that obvious.

The additions on top of that such as which manifests to load and the install state secret are openshift specific and need to finessing via a POC to make sure they're what we want before we go to far down this road